test/e2e/mongodbtests/mongodbtests_tls.go

```diff
@@ -0,0 +1,180 @@
+package mongodbtests
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"testing"
+	"time"
+
+	e2eutil "github.com/mongodb/mongodb-kubernetes-operator/test/e2e"
+	f "github.com/operator-framework/operator-sdk/pkg/test"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	mdbv1 "github.com/mongodb/mongodb-kubernetes-operator/pkg/apis/mongodb/v1"
+	"github.com/stretchr/testify/assert"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+// CreateTLSResources will setup the CA ConfigMap and cert-key Secret necessary for TLS
+// The certificates and keys are stored in testdata/tls
+func CreateTLSResources(mdb *mdbv1.MongoDB, ctx *f.TestCtx) func(*testing.T) {
+	return func(t *testing.T) {
+		tlsConfig := e2eutil.NewTestTLSConfig(false)
+
+		// Create CA ConfigMap
+		ca, err := ioutil.ReadFile("testdata/tls/ca.crt")
+		assert.NoError(t, err)
+
+		configMap := corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      tlsConfig.CAConfigMapName,
+				Namespace: mdb.Namespace,
+			},
+			Data: map[string]string{
+				"ca.crt": string(ca),
+			},
+		}
```

Comment:
remember we have builders we can use for constructing literals

```golang
configMap := configmap.Builder().
    SetName(tlsConfig.CAConfigMapName).
    SetNamespace(mdb.Namespace).
    SetField("ca.crt", string(ca)).
    Build()
```

Reply:
Totally forgot about those, thank you!